PINCache/PINOperationQueue.m

```diff
@@ -35,28 +35,35 @@ @interface PINOperationQueue () {
   NSMutableOrderedSet<PINOperation *> *_highPriorityOperations;
   
   NSMapTable<id<PINOperationReference>, PINOperation *> *_referenceToOperations;
+  NSMapTable<NSString *, PINOperation *> *_identifierToOperations;
 }
 
 @end
 
 @interface PINOperation : NSObject
 
-@property (nonatomic, strong) dispatch_block_t block;
+@property (nonatomic, strong) PINOperationBlock block;
 @property (nonatomic, strong) id <PINOperationReference> reference;
 @property (nonatomic, assign) PINOperationQueuePriority priority;
+@property (nonatomic, strong) dispatch_block_t completion;
+@property (nonatomic, strong) NSString *identifier;
+@property (nonatomic, strong) id data;
 
-+ (instancetype)operationWithBlock:(dispatch_block_t)block reference:(id <PINOperationReference>)reference priority:(PINOperationQueuePriority)priority;
++ (instancetype)operationWithBlock:(PINOperationBlock)block reference:(id <PINOperationReference>)reference priority:(PINOperationQueuePriority)priority identifier:(nullable NSString *)identifier data:(nullable id)data completion:(nullable dispatch_block_t)completion;
 
 @end
 
 @implementation PINOperation
 
-+ (instancetype)operationWithBlock:(dispatch_block_t)block reference:(id<PINOperationReference>)reference priority:(PINOperationQueuePriority)priority
++ (instancetype)operationWithBlock:(PINOperationBlock)block reference:(id<PINOperationReference>)reference priority:(PINOperationQueuePriority)priority identifier:(NSString *)identifier data:(id)data completion:(dispatch_block_t)completion
```

Comment:
I renamed the param to `coallescingData` in the interface of [PINOperationQueue](https://github.com/pinterest/PINCache/pull/141/files#diff-39715a7d2aa6ef3a90d70e7e3ceb4e59R33). However, I think the operation doesn't need to know that its data might be coallesced before execution, so I think we should just call it `data` inside `PINOperation`.